public/js/maho-tree.js

```diff
@@ -0,0 +1,935 @@
+     * @prop {boolean} [mahoTreeNestedFolder=true] - enable the nested folder sortable.js plugin
+     * @prop {boolean} [containDepth=false] - contain draggable nodes to the same depth
+     * @prop {boolean} [rootSortable=true] - allow the root node to be sortable
+     *
+     * @typedef {Object} LazyloadOpts
+     * @prop {string} [dataUrl] - URL to load children from
+     * @prop {string} [nodeParameter='node'] - POST param to send node's ID as
+     * @prop {function(MahoTreeNode, URLSearchParams):null} [onBeforeLoad] - callback before children are loaded
+     * @prop {function(MahoTreeNode, Error):null} [onLoadException] - callback when loading children fails
+     *
+     * @typedef {Object} MahoTreeCssVars
+     * @prop {string} [indent='1.25rem'] - length to indent each node level
+     * @prop {string} [spacing='0.25rem'] - length to space in between each node
+     * @prop {string} [line-style='1px dotted #aaa'] - border style for connecting lines
+     * @prop {string} [outline-style='2px solid #0090FF'] - outline style for selected nodes when `config.selectable.hideInputs=true`
+     * @prop {string} [marker-size='10px'] - size for the [+] and [-] expand icon
+     * @prop {string} [icon-size='16px'] - size for the folder or leaf node icons
+     * @prop {string} [label-gap='0.25rem'] - length between icon, checkbox, and label
+     * @prop {string} [disabled-color='#999'] - text color for disabled nodes
+     * @prop {string} [drop-color='#ccc'] - background color for dropping nodes onto folders
+     *
+     * @param {string} container - the container element's DOM ID
+     * @param {Object} [config] - config options
+     * @param {SelectableOpts|boolean|string} [config.selectable=false] - `true` for default options, string `radio|single|simple|nested`, or object
+     * @param {SortableOpts|boolean|string} [config.sortable=false] - `true` for default options, string for sortable group name, or object
+     * @param {LazyloadOpts|boolean|string} [config.lazyload=false] - `true` for default options, string for dataUrl, or object
+     * @param {boolean} [config.rootVisible=true] - is the root node visible
+     * @param {boolean} [config.noLeafNodes=false] - make all node type folder
+     * @param {boolean} [config.noIcons=false] - make all icons invisible
+     * @param {boolean} [config.varienSetHasChanges] - emit event marking the tab as having changes
+     * @param {MahoTreeCssVars} [config.cssVars] -
+     */
+    constructor(container, config = {}) {
+        const containerEl = document.getElementById(container);
+        if (!containerEl) {
+            throw new Error(`Element with ID ${container} not found in DOM`);
+        }
+
+        this.uniqId = generateRandomString(6);
+
+        this.config = {
+            selectable: false,
+            sortable: false,
+            lazyload: false,
+            rootVisible: true,
+            noLeafNodes: false,
+            noIcons: false,
```

Comment:
```suggestion
            leafNodes: true,
            icons: true,
```

could we use positive configurations I find it very confusing to have a "no-something" set to true

maybe we could call them "showLeafNodes" and "showIcons" if icons/leafNodes are not clear enough?

Reply:
Yeah, good call. I think the variant with "show" is better. Will also rename `rootVisible` to `showRoot` for consistency.

Reply:
Actually, I'm a bit hung up on the leaf node option naming. Is `treatAllNodesAsFolders` too wordy?

Reply:
IMHO if wordy is easier to understand then we should go with wordy :-)